controller/src/beerocks/master/db/db.cpp

```diff
@@ -2475,11 +2475,38 @@ const std::list<sChannelScanResults> &db::get_channel_scan_results(const sMacAdd
 //
 // Client Persistent Data
 //
-bool db::is_client_in_persistent_db(const sMacAddr &mac) { return false; }
+bool db::is_client_in_persistent_db(const sMacAddr &mac)
+{
+    auto client_db_entry = client_db_entry_from_mac(mac);
+
+    return bpl::db_has_entry(type_to_string(beerocks::eType::TYPE_CLIENT), client_db_entry);
+}
 
 bool db::add_client_to_persistent_db(const sMacAddr &mac,
                                      std::unordered_map<std::string, std::string> params)
 {
+    auto db_entry = client_db_entry_from_mac(mac);
+
+    if (!bpl::db_has_entry(std::string("type"), db_entry)) {
```

Comment:
Why not simply:
```suggestion
    if (!bpl::db_has_entry("type", db_entry)) {
```

Reply:
Also, what is the API of `bpl::db_has_entry()`?
What does it mean to `bpl::db_has_entry("type", ...)`?

Reply:
an entry (in UCI correlates to section) has a type and a name - all clients have a section-type `client` and looking them up in DB is by the client mac which is used as its name.
similar to the `db::has_node()` the `bpl::db_has_entry()` returns true if entry exist in DB.
to prevent a situation where a mac of the same client was used to describe something that **is not** of section-type `client` we can also specify expected entry type. (if we do not provide it - no validation of entry-type will be performed)

```
/**
 * @brief find if entry exists
 * 
 * @param[in] entry_type type of the requested entry.
 * @param[in] entry_name name of the requested entry.
 * @return true if entry exists in db, false otherwise
 */
bool db_has_entry(const std::string &entry_type, const std::string &entry_name);
```